http3/server.go

```diff
@@ -486,6 +493,17 @@ func ListenAndServeQUIC(addr, certFile, keyFile string, handler http.Handler) er
 // http.DefaultServeMux is used when handler is nil.
 // The correct Alt-Svc headers for QUIC are set.
 func ListenAndServe(addr, certFile, keyFile string, handler http.Handler) error {
+	return ListenAndServeWithCustomAltSvcPort(addr, certFile, keyFile, handler, 0)
+}
+
+// ListenAndServeWithCustomAltSvcPort listens on the given network address for both, TLS and QUIC
+// connetions in parallel. It returns if one of the two returns an error.
+// http.DefaultServeMux is used when handler is nil.
+// The correct Alt-Svc headers for QUIC are set.
+// customAltSvcPort is used to override the default port value in the Alt-Svc response header.
+// This is useful when a Layer 4 firewall is redirecting UDP traffic and clients must use
+// a port different from the port the QUIC server itself is listening on.
+func ListenAndServeWithCustomAltSvcPort(addr, certFile, keyFile string, handler http.Handler, customAltSvcPort uint32) error {
```

Comment:
I don't think we need an extra function here. Users who need to change the port should just initialize an `http3.Server.`.